### PR TITLE
is_ip_address refactor and minor code cleanup

### PR DESCRIPTION
**Pull request by aldenml**

### CODE REVIEW

File: src/broadcast_socket.cpp [hunk 1]
```diff
@@ -56,10 +56,10 @@ using namespace std::placeholders;
 
 namespace libtorrent
 {
-	bool is_ip_address(char const* host)
+	bool is_ip_address(string_view host)
 	{
 		error_code ec;
-		address::from_string(host, ec);
+		address::from_string(host.data(), ec);
```
Comment (arvidn) [inline src/broadcast_socket.cpp]: what's returned from ``data()`` is not necessarily 0-terminated, so I don't think this is right

Comment (arvidn) [inline src/broadcast_socket.cpp]: there is an overload of ``from_string`` that takes a ``std::string const&`` though, so maybe the best solution for now is to make ``is_ip_address`` take that as its parameter

Comment (aldenml) [inline src/broadcast_socket.cpp]: ok

File: src/session_impl.cpp [hunk 2]
```diff
@@ -633,9 +633,7 @@ namespace aux {
 		update_dht();
 		update_peer_fingerprint();
 		update_dht_bootstrap_nodes();
-#ifndef TORRENT_DISABLE_DHT
 		update_dht_announce_interval();
```
Comment (arvidn) [inline src/session_impl.cpp]: is this change significant? I wouldn't expect this function to do anything when DHT is disabled

Comment (aldenml) [inline src/session_impl.cpp]: not, but just for aesthetics.

File: src/torrent.cpp [hunk 3]
```diff
@@ -5790,8 +5790,8 @@ namespace libtorrent
 			return;
 		}
 
-		bool const is_ip = is_ip_address(hostname.c_str());
-		if (is_ip) a.address(address::from_string(hostname.c_str(), ec));
+		bool const is_ip = is_ip_address(hostname);
+		if (is_ip) a.address(address::from_string(hostname, ec));
```
Comment (arvidn) [inline src/torrent.cpp]: heh, this code is a bit weird (unrelated to your patch). ``is_ip_address()`` just calls ``from_string`` and checks the error, it seems redundant to just call it twice here

Comment (aldenml) [inline src/torrent.cpp]: true

**Comment by arvidn** [comment]
lgtm. just one question about the disable DHT define

**Pull request merged**